rules/aws_policy_heredocs_disallowed.go

```diff
@@ -0,0 +1,97 @@
+package rules
+
+import (
+    "strings"
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/hclsyntax"
+	"github.com/terraform-linters/tflint-plugin-sdk/tflint"
+	"github.com/terraform-linters/tflint-ruleset-aws/project"
+)
+
+// AwsPolicyHeredocsDisallowedRule checks that policies are not defined with Heredoc format
+type AwsPolicyHeredocsDisallowedRule struct {
+	tflint.DefaultRule
+
+	resourceType  string
+	attributeName string
+}
+
+// NewAwsPolicyHeredocsDisallowedRule returns new rule with default attributes
+func NewAwsPolicyHeredocsDisallowedRule() *AwsPolicyHeredocsDisallowedRule {
+	return &AwsPolicyHeredocsDisallowedRule{
+		resourceType:  "aws_iam_policy",
+		attributeName: "policy",
+	}
+}
+
+// Name returns the rule name
+func (r *AwsPolicyHeredocsDisallowedRule) Name() string {
+	return "aws_policy_heredocs_disallowed"
+}
+
+// Enabled returns whether the rule is enabled by default
+func (r *AwsPolicyHeredocsDisallowedRule) Enabled() bool {
+	return false
+}
+
+// Severity returns the rule severity
+func (r *AwsPolicyHeredocsDisallowedRule) Severity() tflint.Severity {
+	return tflint.NOTICE
+}
+
+// Link returns the rule reference link
+func (r *AwsPolicyHeredocsDisallowedRule) Link() string {
+	return project.ReferenceLink(r.Name())
+}
+
+// Check checks if a heredoc block is in use.
+func (r *AwsPolicyHeredocsDisallowedRule) Check(runner tflint.Runner) error {
+
+	path, err := runner.GetModulePath()
+	if err != nil {
+		return err
+	}
+	if !path.IsRoot() {
+		// This rule does not evaluate child modules.
+		return nil
+	}
+
+	files, err := runner.GetFiles()
+	if err != nil {
+		return err
+	}
+	for name, file := range files {
+		if err := r.checkHeredocs(runner, name, file); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (r *AwsPolicyHeredocsDisallowedRule) checkHeredocs(runner tflint.Runner, filename string, file *hcl.File) error {
```

Comment:
As already noted, this needs to filter on AWS resources. But beyond that, it is not correct to simply forbid all heredocs in AWS resources. Take this perfectly valid resource for example:

```tf
resource "aws_s3_bucket_object" "readme" {
  bucket = "your_bucket_name"
  key    = "README.md"
  source = "path/to/file"

  content = <<-EOF
    # Hello world!

    I am a markdown document, embedded in a heredoc
  EOF
}
```

Only resource attributes that accept an IAM policy can be lexed for heredoc tokens. This is probably the hardest part of implementing the rule. You might be able to look at name patterns for those attributes but it's also possible you'd have to keep a static `map[string][]string` of resource names to attributes that accept IAM policies.